add ubuntu xenial packaging files

lsm5 opened the pull request:
Signed-off-by: Lokesh Mandvekar <lsm5@fedoraproject.org>

<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/kubernetes-incubator/cri-o/blob/master/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->

**- What I did**

Added a `debian/` directory which contains deb packaging files (currently meant for Ubuntu Xenial)

**- How I did it**

Magic :)

**- How to verify it**

    $ git archive --format=tar.gz --prefix=build-dir/ HEAD > ../cri-o_1.9.2.orig.tar.gz
    (tarball needs to be in same dir as source AFAIK)
    $ dpkg-buildpackage -us -uc

**- Description for the changelog**

Add packaging files for Ubuntu 16.04 LTS (Xenial)
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->


Currently I have a `xenial` branch on my own fork which I rebase on the latest release tag whenever I need to do a new build.

If this commit gets in, I'm hoping that before cutting a new release, we can try a build of the deb package using the packaging files and bump `debian/changelog` as well, so my work gets reduced :) . As far as *trying* a build before release goes, I guess that could be done via a Makefile target that I had in https://github.com/kubernetes-incubator/cri-o/pull/924

@mrunalp @runcom @rhatdan PTAL

wking commented on debian/control:
```diff
@@ -0,0 +1,30 @@
+    libgpgme11-dev,
+    libseccomp-dev,
+    golang,
+    golang-1.8,
```
Do you need both `golang` and the versioned `golang-1.8`?  I'd expect `golang` would be sufficient.

wking commented on debian/copyright:
```diff
@@ -0,0 +1,9 @@
+Source: https://github.com/kubernetes-incubator/cri-o.git
+Comment:
+ All golang deps bundled.
+ FIles under vendor/ could have their own independent licenses.
```
“FIles” → “Files”.

If a deb package includes vendored Go dependencies, are you really allowed to ignore their licenses?

runcom commented:
+1

TomSweeneyRedHat commented:
Looks like some gofmt whitespace errors @lsm5 lsm5

lsm5 commented:
Closing this in favour of https://github.com/kubernetes-incubator/cri-o/pull/1380 . I fixed the copyright file there and addressed other comments (afaict).
